src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -1397,3 +1482,71 @@ int sd_dhcp_server_set_relay_agent_information(
         free_and_replace(server->agent_remote_id, remote_id_dup);
         return 0;
 }
+
+int sd_dhcp_server_add_static_lease(
+                sd_dhcp_server *server,
+                const struct in_addr* address,
+                uint8_t *client_id,
+                size_t client_id_size) {
+         _cleanup_(dhcp_lease_freep) DHCPLease *lease = NULL, *old = NULL;
+        int r;
+
+        assert_return(server, -EINVAL);
+        assert_return(address, -EINVAL);
+        assert_return(client_id, -EINVAL);
+        assert_return(client_id_size == ETH_ALEN, -EINVAL);
+        assert_return(!sd_dhcp_server_is_running(server), -EBUSY);
+
+        lease = new(DHCPLease, 1);
+        if (!lease)
+                return -ENOMEM;
+
+        *lease = (DHCPLease) {
+                .address = address->s_addr,
```

Comment:
Maybe, it is better to check the address is not used in other static leases by using `static_lease_is_address_taken()` or `static_leases_have_address()`.